corehq/apps/settings/forms.py

```diff
@@ -54,8 +56,19 @@ def __init__(self, user, *args, **kwargs):
             ),
         )
 
-    def clean_new_password1(self):
-        return clean_password(self.cleaned_data.get('new_password1'))
+    def clean_old_password(self):
+        from corehq.apps.hqwebapp.utils import decode_password
+        old_password = decode_password(self.cleaned_data['old_password'])
+        if old_password == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='password_empty',
+            )
+        if not self.user.check_password(old_password):
+            raise forms.ValidationError(
+                self.error_messages['password_incorrect'],
+                code='password_incorrect',
+            )
```

Comment:
I think @millerdev point is that the super method already does the password validation so it shouldn't be necessary to copy that in there. Rather you could just call the super method as he's suggested in his code examples.